mt_distributions/ExponentialDistribution.py

```diff
@@ -0,0 +1,206 @@
+import math
+from .GeneralDistribution import Distribution
+from .ErlangDistribution import Erlang
+
+class HypoExponential(Distribution):
+    """ Hypoexponential Distribution Class for calculating and 
+	visualizing a hypoexponential distribution.
+	
+	Attributes:
+		lp1 (float) representing the decay parameter (λ) of the first composing Exponential Distribution
+        lp2 (float) representing the decay parameter of the second composing Exponential Distribution
+		data_list (list of floats) a list of floats extracted from the data file
+	
+    """
+    def __init__(self, lp1, lp2):
+        self.lp1 = lp1
+        self.lp2 = lp2
+
+        Distribution.__init__(self, self.calculate_mean(), self.calculate_stdev())
+    
+    def calculate_mean(self):
+        """Function to calculate the mean from the lp factors
+        
+        Args: 
+            None
+        
+        Returns: 
+            float: mean of the distribution
+    
+        """
+        self.mean = (1/self.lp1) + (1/self.lp2)
+        return self.mean
+
+    def calculate_variance(self):
+        """Function to calculate the variance from the lp factors
+        
+        Args: 
+            None
+        
+        Returns: 
+            float: variance of the distribution
+    
+        """
+        self.variance = (1/(self.lp1*self.lp1))+(1/(self.lp2*self.lp2))
+        return self.variance
+
+    def calculate_stdev(self):
+        """Function to calculate the standard deviation from the variance
+        
+        Args: 
+            None
+        
+        Returns: 
+            float: standard deviation of the distribution
+    
+        """
+        self.calculate_variance()
+        self.stdev = math.sqrt(self.variance)
+        return self.stdev
+
+    def pdf(self, x):
+        """Probability density function calculator for the distribution.
+        
+        Args:
+            x (float): point for calculating the probability density function
+            
+        
+        Returns:
+            float: probability density function output
+        """
+        coefficient = (self.lp1*self.lp2)/(self.lp1 - self.lp2)
+        return coefficient*(math.exp(-x*self.lp2)-math.exp(-x*self.lp1))
+    
+    def __add__(self, other):
+        raise NotImplementedError
+    
+    def __repr__(self):
+        """Function to output the characteristics of the HypoExponential instance
+		
+		Args:
+			None
+		
+		Returns:
+			string: characteristics of the HypoExponential
+		
+		"""
+        return "Lambda Parameter 1 {}, Lambda Parameter 2 {}".format(self.lp1, self.lp2)
+
+class Exponential(Distribution):
```

Comment:
Just curious, why is this in the same file as HypoExponential class. They don't appear to depend on each other. Do they provide similar functionality to each other?

Reply:
I figured since Hypoexponentials are special distribution conformed of summed exponentials that they were logically related. I'd be happy to separate them and keep the one-file-to-a-class structure, if you prefer!